besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java

```diff
@@ -805,6 +825,19 @@ public BesuController build() {
     if (privacyParameters.getPrivateStorageProvider() != null) {
       closeables.add(privacyParameters.getPrivateStorageProvider());
     }
+    if (preMergeBlockPruningConfiguration.preMergeBlockPruningEnabled()) {
```

Comment:
Would be good to detect if pruning is already completed and skip setting this up